lib/spack/spack/cmd/mirror.py

```diff
@@ -97,6 +98,19 @@ def setup_parser(subparser):
         default=spack.cmd.default_list_scope(),
         help="configuration scope to read from")
 
+    # Binary
+    upload_parser = sp.add_parser('binary', help=mirror_binary.__doc__)
+    upload_parser.add_argument('-d', '--directory', default=None,
```

Comment:
I think this needs to be a bit more abstract. Right now the user can call this with any directory, but it really only makes sense if you call it with a local mirror (e.g. the one we build in `var/spack/cache` as tarballs are downloaded).  I think we need this operation (upload a directory) but it’s lower level — don’t allow users to upload just any directory.